main/lsp/requests/document_formatting.cc

```diff
@@ -72,22 +72,18 @@ void DocumentFormattingTask::preprocess(LSPPreprocessor &preprocessor) {
     auto path = config.remoteName2Local(params->textDocument->uri);
 
     auto maybeFileContents = preprocessor.maybeGetFileContents(path);
-    std::string ignoredFileContents;
-    string_view sourceView;
-    if (maybeFileContents.has_value()) {
-        sourceView = maybeFileContents.value();
-    } else if (sorbet::FileOps::exists(path)) {
-        // If the requested file path isn't in the workspace,
-        // we won't be able to load it, in which case
-        // we leave sourceView as empty and this becomes a no-op
-
-        // In this case, the request is for a file that's
-        // not open in the IDE, so we read it from disk instead.
-        // We store it in `ignoredFileContents` to ensure that the
-        // `sourceView` points at valid data for its lifetime.
-        ignoredFileContents = sorbet::FileOps::read(path);
-        sourceView = ignoredFileContents;
+    if (!maybeFileContents.has_value()) {
+        // If we weren't already tracking the file, don't format it. If we were to load the file from disk and format
+        // it, we would be potentially loading a version that doesn't include any of the changes the client made, and
+        // formatting that version and sending it back would throw away any changes that had been made. In the future if
+        // we decide to track changes on ignored files, we could rethink returning an error here, but for now this is
+        // the safest response.
+        response->error =
+            make_unique<ResponseError>((int)LSPErrorCodes::InvalidRequest, "Unable to format ignored/unknown file");
```

Comment:
Is this going to show up as a popup on save every time the user saves an ignored file?

I imagine that's going to be annoying.

Reply:
In my local testing, there's no popup or interaction with the user -- the file just doesn't get formatted.

Reply:
I also tested with neovim, and found that it also raises no user-visible error.

Reply:
The one place that we do see a message is the `Output` tab for sorbet in VSCode. I think this is fine as it gives an explanation for why formatting didn't happen, but wanted to see if you felt the same. Just to be clear, there's no alert or popup that indicates that formatting failed, just the logged error in the `Output` tab.

Reply:
wfm